### PR TITLE
feat: Add childrenChangedCallback() mixin. Prettier'd some other files.

### PR DESCRIPTION
**Pull request by treshugart**
## Requirements

- [x] The commit message follows [our guidelines](https://github.com/skatej/skatej/blob/master/docs/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bug fixes / features)

## Issues

#995

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 // @flow
```
Comment (treshugart) [inline src/index.js]: Mostly prettier.

File: src/index.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
-export * from "./emit";
-export * from "./link";
-export * from "./with-component";
-export * from "./with-props";
-export * from "./with-renderer";
-export * from "./with-unique";
+export * from './define';
+export * from './emit';
+export * from './link';
+export * from './with-children';
```
Comment (treshugart) [inline src/index.js]: Added export.

File: src/with-children.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// @flow
```
Comment (treshugart) [inline src/with-children.js]: This is the the meat of the PR.

File: src/with-component.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
 // @flow
 
-import { withProps } from "./with-props";
-import { withRenderer } from "./with-renderer";
-import { withUnique } from "./with-unique";
+import { withChildren } from './with-children';
```
Comment (treshugart) [inline src/with-component.js]: Since this is an "all in" mixin, we add it.

File: src/with-props.js [hunk 5]
```diff
@@ -4,15 +4,15 @@ import type {
   HasConstructor,
```
Comment (treshugart) [inline src/with-props.js]: Only prettier.

File: src/with-renderer.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 // @flow
```
Comment (treshugart) [inline src/with-renderer.js]: Only prettier.

File: src/with-unique.js [hunk 7]
```diff
@@ -2,14 +2,15 @@
```
Comment (treshugart) [inline src/with-unique.js]: Only prettier.

File: test/unit/index.spec.js [hunk 8]
```diff
@@ -1,35 +1,37 @@
 /* eslint-env jest */
```
Comment (treshugart) [inline test/unit/index.spec.js]: Mostly prettier.

File: test/unit/index.spec.js [hunk 9]
```diff
@@ -1,35 +1,37 @@
...
-    expect(typeof api.link).toBe("function");
+  it('skate.link', () => {
+    expect(typeof api.link).toBe('function');
   });
 
-  it("skate.with*", () => {
-    expect(typeof api.withComponent).toBe("function", "withComponent");
-    expect(typeof api.withProps).toBe("function", "withProps");
-    expect(typeof api.withRenderer).toBe("function", "withRenderer");
+  it('skate.with*', () => {
```
Comment (treshugart) [inline test/unit/index.spec.js]: Added missing test for `withUnique` and added `withChildren`.

File: test/unit/with-children.spec.js [hunk 10]
```diff
@@ -0,0 +1,58 @@
+// @flow
```
Comment (treshugart) [inline test/unit/with-children.spec.js]: Tests for `withChildren`.

File: test/unit/with-children.spec.js [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+/** @jsx h */
+/* eslint-env jest */
+
+import { mount } from '@skatejs/bore';
+import { h } from '@skatejs/val';
+import { define, withChildren, withUnique } from '../../src';
+
+// Very basic implementation so that the withChildren() mixin works.
+global.MutationObserver = class {
```
Comment (treshugart) [inline test/unit/with-children.spec.js]: I had to do this because Undom doesn't fill this.

File: test/unit/with-children.spec.js [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+
+const Elem = define(
+  class extends withChildren(withUnique()) {
+    childrenChangedCallback() {
+      ++this.called;
+    }
+  }
+);
+
+describe('withChildren', () => {
```
Comment (treshugart) [inline test/unit/with-children.spec.js]: I wanted to add another test here to make sure observers won't double up if the node is removed and reattached, but then we'd be testing the mocked version.

**Comment by Hotell** [comment]
dude those short comments about why did you do what are genius ! 🍺

File: src/with-props.js (hunk 5 above)
Comment (Hotell) [reply src/with-props.js]: we need to automatize prettier stuff via lint-staged

File: test/unit/with-children.spec.js [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+    node.appendChild(<div />);
+    node.appendChild(<div />);
+    mount(node);
+    setTimeout(() => {
+      expect(node.called).toBe(1);
+      done();
+    });
+  });
+
+  it('should be when children are modified', done => {
```
Comment (Hotell) [inline test/unit/with-children.spec.js]: nit `should be when` -> `should be called when`

File: src/with-children.js [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+
+export const withChildren = (
```
Comment (Hotell) [inline src/with-children.js]: 👏

**Pull request merged**